server/modules/survey/service/surveyService.js

```diff
@@ -12,6 +20,93 @@ export const startPublishJob = (user, surveyId) => {
   return job
 }
 
+export const exportSurvey = async ({ surveyId, res }) => {
+  const survey = await SurveyManager.fetchSurveyAndNodeDefsAndRefDataBySurveyId(surveyId)
+
+  const base = '/tmp/export'
+  const prefix = `survey_${surveyId}`
+  const pathPrefix = FileUtils.join(base, prefix)
+  await FileUtils.rmdir(pathPrefix)
+  await FileUtils.mkdir(pathPrefix)
```

Comment:
better not to create a file on disk. what if two users export the same survey at the same time?

you can stream the zip file directly to the response with jszip (below a snippet from fra)

```
const JSZip = require('jszip')
...

try {
      const files = await ExportService.exportData(ExportService.EXPORT_TYPE.CSV, false)

      const zip = new JSZip()

      // Include README.txt in the zipfile

      const readFile = util.promisify(fs.readFile)
      const readmeFileName = 'README.txt'
      const readmeContent = await readFile(path.resolve(__dirname, `./${readmeFileName}`))

      zip.file(readmeFileName, readmeContent)
      Object.values(files).forEach(({ fileName, content }) => zip.file(fileName, content))

      zip
        .generateNodeStream({ type: 'nodebuffer', streamFiles: true })
        .pipe(res)
        .on('finish', function () {
          res.end()
        })
    } catch (err) {
      Request.sendErr(res, err)
    }
```